File: cs50/cash/cash.py
# Project description can be found at: https://cs50.harvard.edu/x/2023/psets/6/cash/

from cs50 import get_float

change_owed = -1
while change_owed < 0:
    change_owed = get_float("Change owed: ")
change_owed *= 100
def quarters(n):
    a = 0
    for i in range(int(n)):
        a += 1
        n -= 25
        if n < 0:
            a -= 1
            break
    return a

def dimes(n):
    b = 0
    for i in range(int(n)):
        b += 1
        n -= 10
        if n < 0:
            b -= 1
            break
    return b

def nickels(n):
    c = 0
    for i in range(int(n)):
        c += 1
        n -= 5
        if n < 0:
            c -= 1
            break
    return c

def pennies(n):
    d = 0
    for i in range(int(n)):
        d += 1
        n -= 1
        if n < 0:
            d -= 1
            break
    return d

quarters = quarters(change_owed)
change_owed = change_owed - quarters * 25

dimes = dimes(change_owed)
change_owed = change_owed - dimes * 10

nickels = nickels(change_owed)
change_owed = change_owed - nickels * 5

pennies = pennies(change_owed)
change_owed = change_owed - pennies * 1

coins = quarters + dimes + nickels + pennies
print(coins)

File: cs50/cash/cash.c
// Project description can be found at: https://cs50.harvard.edu/x/2023/psets/1/cash/

#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);
int a = 0;
int b = -1;
int c = -1;
int d = -1;
int e = -1;

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

//Prompts the user to input the amount of change that is owed
int get_cents(void)
{
    do
    {
        a = get_int("Change Owed: ");
    }
    while (a < 0);
    return a;
}

//Calculates the amount of quarters needed
int calculate_quarters(int cents)
{
    for (int i = 0; i <= cents; i = i + 25)
    {
        b = b + 1;
    }
    return b;
}

//Calculates the amount of dimes needed
int calculate_dimes(int cents)
{
    for (int i = 0; i <= cents; i = i + 10)
    {
        c = c + 1;
    }
    return c;
}

//Calculates the amount of nickels needed
int calculate_nickels(int cents)
{
    for (int i = 0; i <= cents; i = i + 5)
    {
        d = d + 1;
    }
    return d;
}

//Calculates the amount of pennies needed
int calculate_pennies(int cents)
{
    for (int i = 0; i <= cents; i++)
    {
        e = e + 1;
    }
    return e;
}
